TriblerGUI/widgets/trustpage/js/drawing.js

```diff
@@ -28,48 +28,102 @@ function drawNodes(svg, data, on_click) {
     // Always remove existing nodes before adding new ones
     selectNodes(svg).remove();
 
+    // Do not display the label after the graph is updated
+    hoverInfoLabel.style("display", "none");
+
     var selection = selectNodes(svg).data(data.nodes).enter();
 
-    // Create an <svg.node> element.
+    // Create an <svg.node> element
     var groups = selection
         .append("svg")
         .attr("overflow", "visible")
         .attr("class", "node");
 
-    // Append a <circle> element to it.
-    var circles = groups
+    // Append a <circle> element to it
+    groups
         .append("circle")
         .attr("fill", function (d) {
             return getNodeColor(d, data)
         })
         .attr("r", config.node.circle.radius)
         .attr("cx", config.node.circle.cx)
-        .attr("cy", config.node.circle.cy)
-        .style("cursor", config.node.circle.cursor)
-        .on("click", on_click)
-        .on("mouseenter", function () {
-            d3.select(this).transition().ease(d3.easeElasticOut).delay(0).duration(300).attr("r", 25);
-        }).on("mouseout", function () {
-            d3.select(this).transition().ease(d3.easeElasticOut).delay(0).duration(300).attr("r", 20);
-        });
+        .attr("cy", config.node.circle.cy);
 
-    // Append a <text> element to it
+    // Text element in the middle of the node with the last x public key characters
     groups
         .append("text")
-        .attr("x", config.node.publicKeyLabel.x)
-        .attr("y", config.node.publicKeyLabel.y)
+        .attr("dominant-baseline", "central")
+        .attr("text-anchor", "middle")
         .style("font-family", config.node.publicKeyLabel.fontFamily)
         .style("font-size", config.node.publicKeyLabel.fontSize)
+        .style("font-weight", config.node.publicKeyLabel.fontWeight)
         .style("fill", config.node.publicKeyLabel.color)
         .text(function (d) {
-            return d.public_key.substr(-5);
+            return d.public_key.substr(-config.node.publicKeyLabel.characters);
+        });
+
+    // Transparent circle for the mouse events
+    groups
+        .append("circle")
+        .style("fill-opacity", "0")
+        .attr("r", config.node.circle.radius)
+        .attr("cx", config.node.circle.cx)
+        .attr("cy", config.node.circle.cy)
+        .style("cursor", config.node.circle.cursor)
+        .on("click", on_click)
+        .on("mouseover", function (d) {
+            mouseInNode(d, hoverInfoLabel);
+        })
```

Comment:
since you need hoverInfoLabel here, pass it as a parameter to drawNodes, because now you are simply assuming that it is defined somewhere else.